bitcoin/src/address/mod.rs

```diff
@@ -154,7 +154,7 @@ impl fmt::Display for AddressInner {
                     NetworkKind::Main => PUBKEY_ADDRESS_PREFIX_MAIN,
                     NetworkKind::Test => PUBKEY_ADDRESS_PREFIX_TEST,
                 };
-                prefixed[1..].copy_from_slice(&hash[..]);
+                prefixed[1..].copy_from_slice(hash.as_byte_array());
```

Comment:
In dd06be037e3dd11d1caba4836824ca830940803d:

It would be clearer if these changes (using `as_byte_array` throughout) were done in their own commit.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/2718